lib/Sema/TypeCheckDeclOverride.cpp

```diff
@@ -963,7 +963,8 @@ static GenericSignature *getNewGenericSignature(ValueDecl *base,
         GenericSignatureBuilder::FloatingRequirementSource::forAbstract();
 
     unsigned superclassDepth = 0;
-    if (baseClassSig) {
+    
+    if (baseGenericCtx->isGeneric()) {
```

Comment:
I think the old condition was correct. However, `baseGenericCtx->getGenericParams()` should be `baseClassSig->getParams()`.